crates/api/src/systems/event_state_system.rs

```diff
@@ -289,7 +289,7 @@ impl EventStateSystem {
                         current_node,
                         ecm,
                         &theme,
-                        Some(&self.context_provider.event_queue),
+                        Some(self.context_provider.event_adapter.clone()),
```

Comment:
Just for curiosity:  Is this excessive cloning required? Wouldnt make it slow? How many times this function called ?

Reply:
It is not slow, because it just clone on a shared pointer `Arc`. But I will use a reference there. Thx.

Reply:
oh, I thought it would clone the whole EventAdapter struct, sorry